students/brian_warn/session05/except_exercise.py

```diff
@@ -0,0 +1,55 @@
+#!/usr/bin/python
+
+"""
+An exercise in playing with Exceptions.
+Make lots of try/except blocks for fun and profit.
+
+Make sure to catch specifically the error you find, rather than all errors.
+"""
+
+from except_test import fun, more_fun, last_fun
+
+
+# Figure out what the exception is, catch it and while still
+# in that catch block, try again with the second item in the list
+first_try = ['spam', 'cheese', 'mr death']
+
+try:
+    joke = fun(first_try[0])
+except NameError as ne:
+    print("fun() is not coded properly: " + str(ne))
+
+# Here is a try/except block. Add an else that prints not_joke
+try:
+    not_joke = fun(first_try[2])
+except SyntaxError as se:
+    print("Run Away!" + str(se))
+
+
```

Comment:
Note that your SyntaxError block did not run. If you add the else block, you will see there was actually not an error.